### PR TITLE
Code examples for episode 311 & 312

### PR DESCRIPTION
**Pull request by int3l**

### CODE REVIEW

File: sample_code/ep311/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+bash -c 'echo hi && echo hello 1>&2' >& /dev/null
+bash -c 'echo hi && echo hello 1>&2' >& f
+cat f
+
+bash -c 'echo hi && echo hello 1>&2' 1>&2 > f
+bash -c 'echo hi && echo hello 1>&2' 2>&1 > f
+
+# old bash full redirect for both stdout & stderr
+bash -c 'echo hi && echo hello 1>&2' &> f
+bash -c 'echo hi && echo hello 1>&2' > f 2>&1
```
Comment (int3l) [inline sample_code/ep311/README.md:24]: I was also puzzled by this one :D who would of guessed that ordering of redirection syntax matters ...

File: sample_code/ep312/rev06/t.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import os
+
+# from typing import TYPE_CHECKING
+
+if False:  # TYPE_CHECKING:
```
Comment (int3l) [inline sample_code/ep312/rev06/t.py:5]: LoL, very bizarre haX

File: sample_code/ep311/README.md (hunk 1 above)
Comment (asottile) [reply sample_code/ep311/README.md:24]: yeah this was quite annoying!

File: sample_code/ep312/rev06/t.py (hunk 2 above)
Comment (asottile) [reply sample_code/ep312/rev06/t.py:5]: pyflakes has no branch detection so this "works" :D

**Review by asottile — APPROVED** [review]

**Pull request merged**